ir/src/main/scala/org/scalajs/core/ir/Trees.scala

```diff
@@ -965,9 +965,11 @@ object Trees {
 
   }
 
-  /** A hash of a tree (usually a MethodDef). Contains two SHA-1 hashes */
-  final class TreeHash(val treeHash: Array[Byte], val posHash: Array[Byte]) {
-    assert(treeHash.length == 20)
-    assert(posHash.length == 20)
+  /** A hash of a tree (usually a MethodDef).
+   *
+   *  Contains a SHA-1 hash.
+   */
+  final class TreeHash(val hash: Array[Byte]) {
```

Comment:
So, actually, `TreeHash`es are always inside an `Option`, so making them `AnyVal`s would not improve anything. On the other hand, it would slightly degrade our ability to evolve them in binary compatible ways. So I'd say we keep them as is.

Does that make sense?